daemon/prompter.c

```diff
@@ -999,6 +1065,55 @@ prompter_cancel_invocation(prompter_t *self)
     }
 }
 
+static void
+prompter_watch_name(prompter_t *self, GDBusConnection *connection, const gchar *name)
+{
+    if( !g_hash_table_contains(self->prm_watchers, name) ) {
+        g_hash_table_insert(self->prm_watchers, g_strdup(name),
+                            watcher_create(self, connection, name));
```

Comment:
hmm... watcher_create() caches the name it is given as-is vs. copy is used as hash table key -> different and a bit surprising lifetimes, and validity of the other string is under control of whatever calls this function...

I'd either use the same duplicated string for hash key and watcher object (and document nature of wtc_name), or dup within watcher_ctor.

Reply:
Memory management... Yes, I overlooked that a bit. Btw, it's getting interesting when `prompter_handle_name_lost()` is called with `name` that's actually owned by `watcher_t` but that `watcher_t` get's destroyed by `prompter_handle_name_lost()`.